Sources/SourceKitLSP/DocumentManager.swift

```diff
@@ -192,6 +194,35 @@ public final class DocumentManager {
       return document.latestSnapshot
     }
   }
+
+  private func willEditDocument(_ before: LineTable, edit: TextDocumentContentChangeEvent) -> SourceEdit {
+    guard let range = edit.range else {
+      return SourceEdit(
+        range: Range(
+          uncheckedBounds: (
+            AbsolutePosition(utf8Offset: 0), 
+            upper: AbsolutePosition(utf8Offset: before.content.utf8.count)
+          )
+        ),
```

Comment:
Since you’re touching this anyway, let’s not use `uncheckedBounds`. If the bounds aren’t valid, it will just blow up later and make it a lot harder to debug the code. And I don’t think the bounds check is a performance bottleneck.

```suggestion
        range: AbsolutePosition(utf8Offset: 0)..<AbsolutePosition(utf8Offset: before.content.utf8.count)
```

Same a little further below.